### DISCUSSION TITLE
Exporting Model conversions

### DISCUSSION THREAD

**Discussion by UnkoC0wbOy** [ultralytics/hub#12] [Q&A]
How long is it typically taking users to export the trains dmodel to another format such as coreml? I know this is likely to vary according to model size. I've been stuck on export for going on 12 hours for my latest attempt.  Most attempts appear to time out when my computer falls asleep.

**Comment by glenn-jocher** [comment]
@UnkoC0wbOy 15 seconds for YOLOv5s exported to CoreML on Colab.

<img width="1049" alt="Screen Shot 2022-01-14 at 10 03 05 AM" src="https://user-images.githubusercontent.com/26833433/149577849-a3f55ce5-cc9a-4d9a-89e8-f7939ce1cc6f.png">

**Reply by UnkoC0wbOy** [reply]
I would really prefer to just use the hub... I've wasted so much time with the YOLOv5 notebook trying to get a CoreML export that works correctly that I'd just prefer to move on from it at this point until the documentation improves.. The HUB has so much promise as a simplified system.

**Reply by glenn-jocher** [reply]
@UnkoC0wbOy yeah you can definitely use HUB for easy exports too, i.e. screenshot below. For the repo the command should be pretty simple. You can use Colab for export if your local environment is causing problems:

```bash
python export.py --weights yolov5s.pt --include coreml  # export
python detect.py --weights yolov5s.mlmodel  # inference
```

<img width="1629" alt="Screen Shot 2022-01-14 at 7 04 44 PM" src="https://user-images.githubusercontent.com/26833433/149609746-85f603ae-cad0-4546-a068-1194321a8c78.png">

**Reply by UnkoC0wbOy** [reply]
Am I missing something? I have a model trained using "The Hub", but I cannot get it to export and be able to download in any of the other formats other than the default Pytorch format. I am trying to get it converted to CoreML. I will click the Export button in the CoreML box and let it run for hours with seemingly no progress. Does the export happen on my local machine or does it happen in the hub? Is my computer just underpowered? 

I have spent the last week trying to get a working coreml export from the YOLOv5 Google Colab Notebook to no avail... I can export a CoreML Model it just doesn't work correctly....

![Screen Shot 2022-01-21 at 12 31 02 AM](https://user-images.githubusercontent.com/89663444/150486842-cb6034d1-12b6-4a86-aba2-d9189e8a1aeb.png)

My environment details are listed below:

Ultralytics HUB Version
v0.1.10-beta.3
Client User Agent
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.2 Safari/605.1.15
Operating System
MacIntel
Browser Window Size
1717 x 828
Server Timestamp
1642751082

**Reply by glenn-jocher** [reply]
@UnkoC0wbOy thanks for the feedback and your patience! We're working through the CoreML bug this weekend. In the meantime you should be able to download the *.pt model and then export it using the public repo like this:

```python
# Install YOLOv5
!git clone https://github.com/ultralytics/yolov5  # clone
%cd yolov5
%pip install -qr requirements.txt  # install

# Export to CoreML
!python export.py --weights downloaded_model.pt --include coreml
```

**Comment by kalenmike** [comment]
@UnkoC0wbOy The export happens on our servers. We have had some challenges recently with our export functionality which we are currently working on. I will look into what happened with your export to CoreML and see what is needed in your specific case.

**Reply by kalenmike** [reply]
@UnkoC0wbOy We identified some bugs that cropped up in our export functions. They have been resolved and you should now be able to export to CoreML using Ultralytics HUB v0.1.11-beta
